include/seastar/coroutine/exception.hh

```diff
@@ -87,9 +93,10 @@ struct exception {
 /// co_return coroutine::make_exception(std::runtime_error("something failed miserably"));
 /// ```
 template<typename T>
+requires std::same_as<T, std::exception_ptr> || std::derived_from<std::remove_reference_t<T>, std::exception>
 [[nodiscard]]
-exception make_exception(T&& t) {
-    return exception(std::make_exception_ptr(std::forward<T>(t)));
+exception make_exception(T&& t) noexcept {
+    return exception(std::forward<T>(t));
```

Comment:
The `make_exception`/`return_exception` functions don't seem to accept `std::exception_ptr` by copy. For example, this doesn't compile for me:

```c++
    std::exception_ptr eptr;
    (void)make_exception(eptr);
```

Is this intentional? I guess there might be legitimate cases where copying is preferred.

Reply:
I wanted to save an extraneous move, if the constructor was defined as exception(std::exception_ptr). The compiler doesn't seem to optimize it well as far as @denesb observed.  But there was no intention of disallowing passing a value by copy.
I'll fix that in the next version.

Reply:
I think we do want to disallow it. The rationale is that copying an exception_ptr is costly, because it touches its atomic reference counts. On the other hand, if somebody really wants to pass a copy (I don't really know why though), they can do so explicitly by creating a copy and moving it.

Reply:
@psarna it is quite cumbersome to copy an exception_ptr explicitly since make_exception_ptr(ex), misleading as it is, will mask the original exception ptr, so one needs to assign the exception_ptr to copy into a new variable and then move it, yikes... :-\
We do allow copying exception ptrs in many other paths.
As typical use case is an aborted object that keeps the abort reason as an exception_ptr member
and when accessed it returns an exceptional future by copying the exception_ptr.  You don't want to move it in this case since the aborted state is sticky.